Source/Plugins/OrientationAnalysis/OrientationAnalysisFilters/ImportEbsdMontage.cpp

```diff
@@ -0,0 +1,371 @@
+    out << "ColStart: " << m_InputFileListInfo.ColStart << "\n";
+    out << "ColEnd: " << m_InputFileListInfo.ColEnd << "\n";
+
+    m_FilterCache.clear();
+
+    setErrorCondition(-11);
+    notifyErrorMessage(getHumanLabel(), ss, getErrorCondition());
+    return;
+  }
+
+  int32_t numRows = tileLayout2d.size();
+  int32_t numCols = tileLayout2d[0].size();
+  int32_t totalTiles = numRows * numCols;
+  int32_t tilesRead = 0;
+
+  std::array<double, 2> globalTileOrigin = {{0.0, 0.0}};
+
+  std::map<QString, AbstractFilter::Pointer> oldCache = m_FilterCache;
+
+  std::map<QString, AbstractFilter::Pointer> newFilterCache;
+
+  for(const FilePathGenerator::TileRow2D& tileRow2D : tileLayout2d)
+  {
+    globalTileOrigin[0] = 0.0; // Reset the X Coord back to Zero for each row.
+    double tileHeight = 0.0;
+    for(const FilePathGenerator::Tile2D& tile2D : tileRow2D)
+    {
+      QFileInfo fi(tile2D.FileName);
+      QString fname = fi.completeBaseName();
+
+      if(m_InputFileListInfo.FileExtension == Ebsd::Ang::FileExt)
+      {
+        readEbsdFile<ReadAngData>(this, tile2D.FileName, m_FilterCache, newFilterCache);
+      }
+      if(m_InputFileListInfo.FileExtension == Ebsd::Ctf::FileExt)
+      {
+        readEbsdFile<ReadCtfData>(this, tile2D.FileName, m_FilterCache, newFilterCache);
+      }
+      if(getErrorCondition() >= 0)
+      {
+        DataContainer::Pointer dc = getDataContainerArray()->getDataContainer(fname);
+        ImageGeom::Pointer imageGeom = dc->getGeometryAs<ImageGeom>();
+
+        std::array<size_t, 3> dims = {{0, 0, 0}};
+        std::tie(dims[0], dims[1], dims[2]) = imageGeom->getDimensions();
+        std::array<float, 3> res = {{0.0f, 0.0f, 0.0f}};
+        std::tie(res[0], res[1], res[2]) = imageGeom->getResolution();
+        std::array<float, 3> origin = {{0.0f, 0.0f, 0.0f}};
+        std::tie(origin[0], origin[1], origin[2]) = imageGeom->getOrigin();
+
+        //
+        origin[0] = globalTileOrigin[0];
+        origin[1] = globalTileOrigin[1];
+        imageGeom->setOrigin(origin.data());
+
+        // Now update the globalTileOrigin values
+        globalTileOrigin[0] += origin[0] + (dims[0] * res[0]);
+        tileHeight = (dims[1] * res[1]);
+
+        if(getCancel())
+        {
+          return;
+        }
+        tilesRead++;
+        if(!getInPreflight())
+        {
+          QString msg = QString("==> [%1/%2] %3").arg(tilesRead).arg(totalTiles).arg(tile2D.FileName);
+          notifyStatusMessage(getHumanLabel(), msg);
+        }
+      }
+    }
+    globalTileOrigin[1] += tileHeight;
+  }
+
+  m_FilterCache = newFilterCache; // Swap our maps. This dumps any previous instantiations of the reader filter that are not used any more.
+  setWarningCondition(0);
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void ImportEbsdMontage::flushCache()
```

Comment:
This function doesn't do anything.  Is that what you intended?

Reply:
Nope. Left overs. Not needed since the filter is relying on the sub-filter to do the caching.